### PR TITLE
We were ignoring the "closed" property on paths.

### PR DESCRIPTION
**Pull request by simeoncran**
In most cases, a closed path is simply one that ends where it starts, so we just left all our paths open and relied on After Effects to complete the path.
But that was wrong in some cases - specifically the closing of the path determines whether a miter or an end cap is used where the end meets the start.

### CODE REVIEW

File: source/LottieData/Optimization/Optimizer.cs [hunk 1]
```diff
@@ -101,13 +103,18 @@ public Animatable<Sequence<BezierSegment>> GetOptimized(Animatable<Sequence<Bezi
...
             var hacked = optimized.KeyFrames.SelectToSpan(pg => HackPathGeometry(pg));
-            return new Animatable<Sequence<BezierSegment>>(hacked[0].Value, hacked, optimized.PropertyIndex);
+            return new Animatable<PathGeometry>(hacked[0].Value, hacked, optimized.PropertyIndex);
         }
 
-        static KeyFrame<Sequence<BezierSegment>> HackPathGeometry(KeyFrame<Sequence<BezierSegment>> value)
-        {
-            return new KeyFrame<Sequence<BezierSegment>>(value.Frame, new Sequence<BezierSegment>(new[] { value.Value.Items[0] }), Vector3.Zero, Vector3.Zero, value.Easing);
-        }
+        // Returns a KeyFrame<PathGeometry> that contains only the first bezier segment of the given
```
Comment (derobinsMSFT) [inline source/LottieData/Optimization/Optimizer.cs:109]: >b [](start = 73, length = 1)

Capitalize.

File: source/LottieData/Optimization/Optimizer.cs [hunk 2]
```diff
@@ -101,13 +103,18 @@ public Animatable<Sequence<BezierSegment>> GetOptimized(Animatable<Sequence<Bezi
...
+        // KeyFrame<PathGeometry>
```
Comment (derobinsMSFT) [inline source/LottieData/Optimization/Optimizer.cs:110]: >> [](start = 32, length = 1)

Period.

File: source/LottieToWinComp/LottieToWinCompTranslator.cs [hunk 3]
```diff
@@ -2055,10 +2055,12 @@ CanvasGeometry CreateWin2dPathGeometry(Path path, ShapeFill.PathFillType fillTyp
                         }
                     }
 
-                    // Leave the figure open. If Lottie wanted it closed it will have defined
-                    // a final bezier segment back to the start.
-                    // Closed simply tells D2D to synthesize a final segment.
-                    builder.EndFigure(CanvasFigureLoop.Open);
+                    // Closed tells D2D to synthesize a final segment. In many cases Closed
+                    // will have no effect because After Effects will have included the final
+                    // segment however it can make a difference, for example if the figure is
```
Comment (derobinsMSFT) [inline source/LottieToWinComp/LottieToWinCompTranslator.cs:2060]: >segment [](start = 23, length = 7)

Nit: semicolon or period.

File: source/LottieData/PathGeometry.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Toolkit.Uwp.UI.Lottie.LottieData
+{
+    /// <summary>
+    /// A sequence of <see cref="BezierSegment"/> that describes the shape of a path.
```
Comment (derobinsMSFT) [inline source/LottieData/PathGeometry.cs:11]: >> [](start = 48, length = 1)

"s"

File: source/LottieToWinComp/LottieToWinCompTranslator.cs (hunk 3 above)
Comment (simeoncran) [reply source/LottieToWinComp/LottieToWinCompTranslator.cs:2060]: You need to explain to me why. This is similar to one we were discussing previously but I don't understand the rule you are applying and why it matters.

---
In reply to: [430660286](https://github.com/windows-toolkit/Lottie-Windows/pull/268#discussion_r430660286) [](ancestors = 430660286)

Comment (derobinsMSFT) [reply source/LottieToWinComp/LottieToWinCompTranslator.cs:2060]: In this case, the sentence strikes me as a run-on/a tad long. Here would be a good spot to split it.

---
In reply to: [430694315](https://github.com/windows-toolkit/Lottie-Windows/pull/268#discussion_r430694315) [](ancestors = 430694315,430660286)

Comment (simeoncran) [reply source/LottieToWinComp/LottieToWinCompTranslator.cs:2060]: Got it. I'll simplify it a bit.

---
In reply to: [430700091](https://github.com/windows-toolkit/Lottie-Windows/pull/268#discussion_r430700091) [](ancestors = 430700091,430694315,430660286)

**Pull request merged**